### ISSUE TITLE
Floating IP assignment is lost when performing blue/green deployment

### ISSUE THREAD

**Issue by dave-lew99** [terraform-providers/terraform-provider-digitalocean#147]
Hi there,

### Terraform Version
Terraform v0.11.8

### Affected Resource(s)
Please list the resources as a list, for example:
- digitalocean_droplet
- digitalocean_floating_ip_assignment

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "digitalocean_droplet" "test-1" {
    image = "centos-7-x64"
    name = "test-1"
    region = "fra1"
    size = "s-1vcpu-1gb"
    private_networking = false
    lifecycle {
        create_before_destroy = true
    }
}

resource "digitalocean_floating_ip" "testip" {
  region     = "fra1"

}
resource "digitalocean_floating_ip_assignment" "testip-assignment" {
  ip_address = "${digitalocean_floating_ip.testip.id}"
  droplet_id = "${digitalocean_droplet.test-1.id}"
    lifecycle {
        create_before_destroy = true
    }
}
```

### Panic Output
If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`.

### Expected Behavior
When updating a droplet with a new image, the new droplet should be provisioned then the floating IP assignment is changed over to the new droplet, before destroying the old droplet.

### Actual Behavior
The floating IP assignment briefly swaps to the newly provisioned instance before being destroyed entirely. Running `terraform apply` again fixes the association.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Change the `image` value
3. Run `terraform apply`

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- Maybe GH-99

**Comment by andrewsomething** [comment]
Thank you for reporting this issue. I believe we have determined the cause and have a fix in place. I'll update here once we've cut a new release.

**Comment by andrewsomething** [comment]
This fix was released as version 1.0.2. You can upgrade with:

    terraform init -upgrade

Thanks again for the report, and please let us know if you hit any other problems.
